### ISSUE TITLE
[data grid] Column resizing doesn't work from group

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-x#13247]
### Steps to reproduce

Link to live example: (required)

Steps:
1. Open https://mui.com/x/react-data-grid/column-groups/#define-column-groups
2. Try to resize the column group

https://github.com/mui/mui-x/assets/3165635/e82b536e-ce40-42d6-9d13-a75ea71a50f3

### Current behavior

Can't resize

### Expected behavior

Can resize

### Context

It was clearer in v6, at least it wouldn't give the end-users visual clue that it's possible to resize: https://codesandbox.io/p/sandbox/distracted-mahavira-zx7924?file=%2Fsrc%2FDemo.tsx%3A12%2C30. I suspect it got regressed on in https://github.com/mui/mui-x/pull/12420/files#r1614695294.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: -

**Comment by michelengelen** [comment]
Have added it to the board! 👍🏼

**Comment by cherniavskii** [comment]
Indeed, it's a regression comparing to v6: https://codesandbox.io/p/sandbox/shy-dream-p5qgk6

**Comment by michelengelen** [comment]
Let us consolidate this one. The issue had two parts, and they resolved differently:

- The regression from the video — a resize separator rendered on group headers, signaling an affordance that does nothing — was fixed by #13455 and released in [v7.7.0](https://github.com/mui/mui-x/releases/tag/v7.7.0). Column groups no longer show resize separators, matching the v6 behavior.
- Actually resizing a group from its header is a feature we do not support yet. Group headers are explicitly non-resizable today. That request is tracked in #5882, including the open design question of how a group resize distributes across the children's `minWidth`/`maxWidth` constraints.

I will close this issue as a duplicate of #5882. Please upvote and share use cases there if you need group resizing.
